Flaky test TestDistributedSpilledQueriesWithTempStorage.testRowNumberLimit

TestDistributedSpilledQueriesWithTempStorage.testRowNumberLimit failed with a Verify exception in SpillableGroupedTopNBuilder.  It looks like it could be a race condition in the operator, and not just a test issue. 

Stack trace:
```
2024-03-06T20:55:24.6586010Z 2024-03-06T14:55:24.645-0600	WARN	TestNG-test=Surefire test-1	com.facebook.presto.testng.services.LogTestDurationListener	Tests from com.facebook.presto.execution.resourceGroups.db.TestQueuesDb took 3.65m
2024-03-06T20:55:25.8949861Z [ERROR] Tests run: 3103, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 2,395.581 s <<< FAILURE! - in TestSuite
2024-03-06T20:55:25.8952283Z [ERROR] com.facebook.presto.tests.TestDistributedSpilledQueriesWithTempStorage.testRowNumberLimit  Time elapsed: 0.287 s  <<< FAILURE!
2024-03-06T20:55:25.8954419Z java.lang.RuntimeException: com.google.common.base.VerifyException
2024-03-06T20:55:25.8956190Z 	at com.facebook.presto.tests.AbstractTestingPrestoClient.execute(AbstractTestingPrestoClient.java:124)
2024-03-06T20:55:25.8958566Z 	at com.facebook.presto.tests.DistributedQueryRunner.execute(DistributedQueryRunner.java:733)
2024-03-06T20:55:25.8960274Z 	at com.facebook.presto.tests.DistributedQueryRunner.execute(DistributedQueryRunner.java:701)
2024-03-06T20:55:25.8962076Z 	at com.facebook.presto.tests.AbstractTestQueryFramework.computeActual(AbstractTestQueryFramework.java:134)
2024-03-06T20:55:25.8964071Z 	at com.facebook.presto.tests.AbstractTestQueryFramework.computeActual(AbstractTestQueryFramework.java:129)
2024-03-06T20:55:25.8966220Z 	at com.facebook.presto.tests.AbstractTestQueries.testRowNumberLimit(AbstractTestQueries.java:1738)
2024-03-06T20:55:25.8967670Z 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2024-03-06T20:55:25.8968925Z 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2024-03-06T20:55:25.8970662Z 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2024-03-06T20:55:25.8971916Z 	at java.lang.reflect.Method.invoke(Method.java:498)
2024-03-06T20:55:25.8973282Z 	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:135)
2024-03-06T20:55:25.8974882Z 	at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:673)
2024-03-06T20:55:25.8976340Z 	at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:220)
2024-03-06T20:55:25.8977798Z 	at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
2024-03-06T20:55:25.8979322Z 	at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:945)
2024-03-06T20:55:25.8980974Z 	at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:193)
2024-03-06T20:55:25.8982747Z 	at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
2024-03-06T20:55:25.8984318Z 	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
2024-03-06T20:55:25.8985728Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2024-03-06T20:55:25.8987442Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2024-03-06T20:55:25.8989118Z 	at java.lang.Thread.run(Thread.java:750)
2024-03-06T20:55:25.8990286Z Caused by: com.google.common.base.VerifyException
2024-03-06T20:55:25.8992639Z 	at com.google.common.base.Verify.verify(Verify.java:100)
2024-03-06T20:55:25.8994161Z 	at com.facebook.presto.operator.SpillableGroupedTopNBuilder.finishMemoryRevoke(SpillableGroupedTopNBuilder.java:232)
2024-03-06T20:55:25.8996105Z 	at com.facebook.presto.operator.TopNRowNumberOperator.finishMemoryRevoke(TopNRowNumberOperator.java:308)
2024-03-06T20:55:25.8997937Z 	at com.facebook.presto.operator.Driver.checkOperatorFinishedRevoking(Driver.java:563)
2024-03-06T20:55:25.8999335Z 	at com.facebook.presto.operator.Driver.handleMemoryRevoke(Driver.java:551)
2024-03-06T20:55:25.9000594Z 	at com.facebook.presto.operator.Driver.processInternal(Driver.java:394)
2024-03-06T20:55:25.9001856Z 	at com.facebook.presto.operator.Driver.lambda$processFor$10(Driver.java:319)
2024-03-06T20:55:25.9003283Z 	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:740)
2024-03-06T20:55:25.9004429Z 	at com.facebook.presto.operator.Driver.processFor(Driver.java:312)
2024-03-06T20:55:25.9005877Z 	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1079)
2024-03-06T20:55:25.9007730Z 	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:165)
2024-03-06T20:55:25.9009606Z 	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:621)
2024-03-06T20:55:25.9011079Z 	at com.facebook.presto.$gen.Presto_null__testversion____20240306_204739_37084.run(Unknown Source)
2024-03-06T20:55:25.9012020Z 	... 3 more
```

@shrinidhijoshi  can you please take a look?

--- a/presto-main/src/main/java/com/facebook/presto/operator/TopNRowNumberOperator.java
+++ b/presto-main/src/main/java/com/facebook/presto/operator/TopNRowNumberOperator.java
@@ -296,9 +296,6 @@ public class TopNRowNumberOperator
     @Override
     public ListenableFuture<?> startMemoryRevoke()
     {
-        if (finishing) {
-            return NOT_BLOCKED;
-        }
         return groupedTopNBuilder.startMemoryRevoke();
     }
 

